x-pack/plugin/profiling/src/main/java/org/elasticsearch/xpack/profiling/action/TransportGetStackTracesAction.java

```diff
@@ -351,17 +366,34 @@ private void searchEventGroupedByStackTrace(
                 new TermsAggregationBuilder("group_by")
                     // 'size' specifies the max number of host ID we support per request.
                     .size(MAX_TRACE_EVENTS_RESULT_SIZE)
-                    .field("process.executable.name")
-                    // missing("") is used to include documents where the field is missing.
-                    .missing("")
+                    .field("Stacktrace.sampling_frequency")
+                    // missing(DEFAULT_SAMPLING_RATE) is used to include documents where the field is missing.
+                    .missing((long) DEFAULT_SAMPLING_FREQUENCY)
                     // 'execution_hint: map' skips the slow building of ordinals that we don't need.
                     // Especially with high cardinality fields, this makes aggregations really slow.
                     .executionHint("map")
-                    .subAggregation(groupByThreadName)
+                    .subAggregation(groupByExecutableName)
+                    .subAggregation(new SumAggregationBuilder("total_count").field("Stacktrace.count"))
             )
             .addAggregation(new SumAggregationBuilder("total_count").field("Stacktrace.count"))
             .execute(handleEventsGroupedByStackTrace(submitTask, client, responseBuilder, submitListener, searchResponse -> {
-                long totalCount = getAggValueAsLong(searchResponse, "total_count");
+                long maxSamplingFrequency = 0;
+
```

Comment:
Can you add a comment here describing the following two operations: First finding the maximum frequency and then upscaling the counts of the lower frequency buckets? It may not be immediately obvious to someone reading the code.